### PR TITLE
Added a WrappedLauncher

### PR DESCRIPTION
**Pull request by WardLT**
Designed to standardize launching tasks from within containers, etc

### CODE REVIEW

File: parsl/launchers/launchers.py [hunk 1]
```diff
@@ -27,6 +31,29 @@ def __call__(self, command, tasks_per_node, nodes_per_block):
...
+        self.prepend = prepend
+
+    def __call__(self, command, tasks_per_node, nodes_per_block):
+        if tasks_per_node > 1:
+            logger.warning('WrappedLauncher ignores the number of tasks per node. '
+                           'You may be getting fewer workers than expected')
+        if nodes_per_block > 1:
+            logger.warning('WrappedLauncher ignores the number of nodes per block. '
+                           'You may be getting fewer workers than expected')
+        return f"{self.prepend or ''} {command}"
```
Comment (benclifford) [inline parsl/launchers/launchers.py]: you can't use f-strings because they are newer than python 3.5

File: parsl/launchers/launchers.py [hunk 2]
```diff
@@ -27,6 +31,29 @@ def __call__(self, command, tasks_per_node, nodes_per_block):
         return command
 
 
+class WrappedLauncher(Launcher):
+    """Wraps the command by prepending commands before a user's command
+
+    As an example, the wrapped launcher can be used to launch a command
+    inside a docker contain by prepending the proper docker invocation"""
+
+    def __init__(self, prepend: Optional[str]=None):
```
Comment (benclifford) [inline parsl/launchers/launchers.py]: I think `prepend` should be non-optional.

File: parsl/tests/sites/test_launchers.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+def test_wrapped_launcher(caplog):
+    w = WrappedLauncher('docker')
+    assert w('test', 1, 1) == 'docker test'
+
+    # Make sure it complains if you have > 1 node or task per node
+    w('test', 2, 2)
+    assert 'tasks per node' in caplog.text
+    assert 'nodes per block' in caplog.text
+
+
```
Comment (benclifford) [inline parsl/tests/sites/test_launchers.py]: parsl/tests/sites/test_launchers.py:13:1: W391 blank line at end of file

Comment (WardLT) [reply parsl/tests/sites/test_launchers.py]: Oops!

**Pull request merged**